src/controllers/projectController.ts

```diff
@@ -0,0 +1,144 @@
+            const projectName = path.basename(projectFilePath, path.extname(projectFilePath));
+
+            // Construct build arguments based on project type
+            const buildDirPath = this.getBuildDirPath();
+            const buildArgs = this.constructBuildArguments(
+                buildDirPath,
+                projectProperties.projectStyle,
+            );
+
+            // Create task arguments
+            const args: string[] = [constants.build, projectFilePath, ...buildArgs];
+
+            // Create task definition
+            const taskDefinition: vscode.TaskDefinition = {
+                type: constants.sqlProjBuildTaskType,
+                label: `Build ${projectName}`,
+                command: constants.dotnet,
+                args: args,
+                problemMatcher: constants.msBuildProblemMatcher,
+            };
+
+            // Create the build task
+            const buildTask = new vscode.Task(
+                taskDefinition,
+                vscode.TaskScope.Workspace,
+                taskDefinition.label,
+                taskDefinition.type,
+                new vscode.ShellExecution(taskDefinition.command, args, { cwd: projectDir }),
+                taskDefinition.problemMatcher,
+            );
+
+            // Execute the task and wait for completion
+            await this.executeBuildTask(buildTask, projectName);
+
+            // Return the DACPAC output path
+            return projectProperties.dacpacOutputPath;
+        } catch (error) {
+            // Send error telemetry
+            sendErrorEvent(
+                TelemetryViews.SqlProjects,
+                TelemetryActions.BuildProject,
+                error instanceof Error ? error : new Error(String(error)),
+                false,
+            );
+            throw error;
+        }
+    }
+
+    /**
+     * Gets the path to the BuildDirectory folder in the extension
+     */
+    private getBuildDirPath(): string {
+        const extensionPath =
+            vscode.extensions.getExtension(constants.extensionId)?.extensionPath ?? "";
+        return path.join(extensionPath, "BuildDirectory");
+    }
+
+    /**
+     * Constructs the build arguments for building a sqlproj file
+     * @param buildDirPath Path to the SQL Tools Service directory containing build dependencies
+     * @param projectStyle The project style (SDK-style or Legacy-style)
+     * @returns An array of arguments to be used for building the sqlproj file
+     */
+    private constructBuildArguments(
+        buildDirPath: string,
+        projectStyle: mssql.ProjectType,
+    ): string[] {
+        const args: string[] = ["/p:NetCoreBuild=true", `/p:SystemDacpacsLocation=${buildDirPath}`];
+
+        // Adding NETCoreTargetsPath only for non-SDK style projects
+        // ProjectType.SdkStyle = 0, ProjectType.LegacyStyle = 1
+        const isSdkStyle = projectStyle === 0;
+
+        if (!isSdkStyle) {
```

Comment:
Why not just do `if(projectStyle === mssql.ProjectType.LegacyStyle) {}`?

Reply:
getting circular dependency issue when using the mssql.ProjectType enum while runtime. 

SQL Database Projects extension declares a dependency on vscode-mssql extension
If projectController.ts uses regular import (import * as mssql), it tries to activate the vscode-mssql extension at load time
But vscode-mssql extension might not be fully activated yet
. This causes the activation failure: "dependency failed to activate" The solution is to keep import type to avoid triggering activation and use the numeric literal 1 for the enum value.